railties/test/generators/multi_db_generator_test.rb

```diff
@@ -11,8 +11,8 @@ def test_multi_db_skeleton_is_created
     run_generator
     assert_file "config/initializers/multi_db.rb" do |record|
       assert_match(/Multi-db Configuration/, record)
-      assert_match(/config.active_record.database_resolver/, record)
-      assert_match(/config.active_record.shard_resolver/, record)
+      assert_match(/middleware.use ActiveRecord::Middleware::DatabaseSelector, resolver, context, options/, record)
+      assert_match(/middleware.use ActiveRecord::Middleware::ShardSelector, resolver, options/, record)
```

Comment:
Is it worth adding a test similar to https://github.com/rails/rails/issues/45162#issuecomment-1145424714 to ensure this works as expected?

Reply:
Probably but I am not excited about how hard that's going to be 😓 I'm not entirely sure it's possible without a huge amount of setup.

Reply:
Actually - I think I just need to test the middleware is installed and can assume that if it is then everything else works. Active Record is already testing the functionality of the middleware itself. I was assuming this was going to be hard because of booting a server, running requests, and checking the connection, but we don't need to do all that.

Reply:
Oh yeah that's what I meant. You could just copy-paste those tests and change the config lines.